src/core/credrank/personalAttribution.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+
+import findLast from "lodash.findlast";
+import {type IdentityId, identityIdParser} from "../identity/id";
+import type {TimestampMs} from "../../util/timestamp";
+import * as C from "../../util/combo";
+
+export type PersonalAttributionProportion = {|
+  +timestampMs: TimestampMs,
```

Comment:
How do we express that Cred is flowing for a fixed period, e.g. the month of april 2021? It feels like having separate `startTimestampMs` and `endTimestampMs` would be nice, although that invites the question of what to do when the data has specified overlapping periods (do they sum? replace each other somehow?). We could also have "latest timestamp wins" semantics.